### DISCUSSION TITLE
Why does w-full reset alignment?

### DISCUSSION THREAD

**Discussion by ponponon** [zauberzeug/nicegui#2187] [Q&A] [answered]
### Question

```python
from nicegui import ui

with ui.row():
    with ui.card().style('minWidth: 600px'):
        with ui.column().classes('w-full items-center'):
            with ui.column():
                ui.markdown(
                    '### title',
                ).style('border-bottom: 1px solid #e5e9ef;')

ui.run(port=8083)

```

<img width="730" alt="图片" src="https://github.com/zauberzeug/nicegui/assets/38725104/d7be9768-7644-4f2f-9a19-c1b97ffae056">

At this point the title is centre-aligned

But if I add `.classes('w-full')` to the internal `ui.column()` it loses the alignment.

```python
from nicegui import ui

with ui.row():
    with ui.card().style('minWidth: 600px'):
        with ui.column().classes('w-full items-center'):
            with ui.column().classes('w-full'):
                ui.markdown(
                    '### title',
                ).style('border-bottom: 1px solid #e5e9ef;')

ui.run(port=8083)

```

<img width="809" alt="图片" src="https://github.com/zauberzeug/nicegui/assets/38725104/ef234ebf-ff4d-4777-aeca-41fd64df5711">

Why is that? Is it a feature of Tailwind CSS?

----

Could it be because `.classes('items-center')` only works on child elements and is not inherited all the way down by the children of the child element?

**Comment by falkoschindler** [comment, marked as answer]
I recommend using some background color for debugging layout problems. Here is a simplified version of your code:

```py
with ui.card().classes('w-[600px] items-center'):
    with ui.column().classes('bg-blue-100 w-full'):
        ui.label('Text')

with ui.card().classes('w-[600px] items-center'):
    with ui.column().classes('bg-blue-100'):
        ui.label('Text')
```

<img width="618" alt="Screenshot 2023-12-16 at 11 15 48" src="https://github.com/zauberzeug/nicegui/assets/5767091/502003cc-5a12-4e67-8cb1-8e06d8744ede">

As you can see, `w-full` lets the column grow to the full width of its surrounding container. But within the column the text is not centered. Therefore it is not centered in the card.

**Reply by ponponon** [reply]
Very good answer, gave me a lot of clues, I already understand why
